### PR TITLE
DOC: revise plotting with folium example

### PR DESCRIPTION
**Pull request by andreaseliasson**
Revision includes:

- Sentence level changes
- Typos
- Removal of redundant info

ref #1896

### CODE REVIEW

File: doc/source/gallery/plotting_with_folium.ipynb [hunk 1]
```diff
@@ -47,44 +54,19 @@
...
-    ]
-   },
-   "outputs": [],
-   "source": [
-    "fig, ax = plt.subplots(figsize=(24,18))\n",
-    "world.plot(ax=ax, alpha=0.4, color='grey')\n",
-    "geo_df.plot(column='Type', ax=ax, legend=True)\n",
-    "plt.title('Volcanoes')"
-   ]
-  },
```
Comment (martinfleis) [inline doc/source/gallery/plotting_with_folium.ipynb:81]: This cell contains a thumbnail used in the gallery. Can you keep it?

File: doc/source/gallery/plotting_with_folium.ipynb [hunk 2]
```diff
@@ -93,41 +75,16 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Stamen Terrain\n",
```
Comment (martinfleis) [inline doc/source/gallery/plotting_with_folium.ipynb:96]: I find the current overview of tiles quite helpful, do you think there's a need to remove them?

**Review by martinfleis — COMMENTED** [review]
Thanks, it is indeed cleaner now, but some removals may be excessive I think. Esp. the one which breaks gallery thumbnail. See the comments in the code.

Just a side note: This will mostly become obsolete once we merge our internal wrapper around folium currently developed here: https://github.com/martinfleis/geopandas-view

File: doc/source/gallery/plotting_with_folium.ipynb (hunk 1 above)
Comment (andreaseliasson) [reply doc/source/gallery/plotting_with_folium.ipynb:81]: Thanks for your comments. Of course, I will keep this!

**Comment by martinfleis** [comment]
Thanks @andreaseliasson!

**Pull request merged**